### ISSUE TITLE
Add version changed for API changes

### ISSUE THREAD

**Issue by taldcroft** [astropy/astropy#8651]
I find it useful (as done in e.g. numpy) to have the version when things changed or were added in the narrative / API docs. E.g. in relevant places we would see a comment like "Default time scale for epoch times like J2000 changed from UTC to TT in version 3.2". The point being that the change log is not where most people wind up when learning about a function or method.

@mhvk points out this is done with a sphinx directive `.. versionchanged:` in numpy.  Is that something astropy could just start using as a matter of policy?

(This was a by-product of discussion in #8600).

EDIT by @pllim and @bsipocz : Preferably, this is to be done in small batches, like one PR for a single version release for a single subpackage. And only as far as as 2.0 and not earlier.

**Comment by bsipocz** [comment]
Yes please, I very much appreciate this info in the API docs, makes it so much easier to track down when something was introduced.

I would also suggest that adding this to the API docs retrospect this is an excellent `package-novice` issue implementation wise but a good way to get the to know the codebase and dev process.

**Comment by pllim** [comment]
Since this is marked as "novice" and "good first issue", would be nice to give some ideas on how to `grep` for the doc texts that need this extra directive.

**Comment by bolt25** [comment]
I'm new here and would like to start contributing, can anyone help me get started with the process?

**Comment by nikita-0209** [comment]
Hi, I am a student. I want to work on this issue as a step to familiarise the project. Can anyone guide me on how to access and edit the API docs?

**Comment by mhvk** [comment]
@nikita-0209 - I'm not sure this is quite appropriately labelled as `novice`, but the idea would be that you go through `CHANGES.rst`, starting at the version now being prepared for release (4.1), check where new features are introduced and then add `.. versionchanged: 4.1` in the relevant docstring of the function or class.  I would start with just making a PR for a single case, as we may well have to introduce that actual directive.

I should add that since we are just before feature freeze, few of us will have time to help right now, so if the above does not suffice, please wait for a few days and ping again early next week...

**Comment by bsipocz** [comment]
@mhvk - this is labelled as package-novice as one needs only go through the changelog/history and add a comment for the docstrings, so not no internal knowledge of astropy is needed.

**Comment by shubham00jain** [comment]
@mhvk @astrojuanlu Hi! I would like to work on this issue. I saw in `changes.rst` that `astropy.config` was listed under New Features, now do I need to add `.. versionchanged: 4.1` or `.. versionchanged: 4.2` ?And in which file do I need to add this? Thanks!

**Comment by tactipus** [comment]
Hi everyone! @astrojuanlu @shubham00jain @mhvk @taldcroft still need help on this?

**Comment by NathanMuniz18** [comment]
hello guys, can i try help on this?

**Comment by astrofrog** [comment]
At this point the package is large enough and there have been enough versions that we will probably need to find a way to automate this rather than doing by hand.

**Comment by NathanMuniz18** [comment]
It's my first time contributing in open source, how can I help you guys in this issue?

**Comment by astrofrog** [comment]
A good place to start would be to go through the docs/changes api entries and adding the versionadded: 7.0.0 directive to the docstring of any new function and making sure it renders ok if you build the documentation locally.

**Comment by tactipus** [comment]
Morning, peeps. I'm happy to help!

**Comment by sn82978** [comment]
Hello! I'm new here, I was wondering what I can do to help with this issue?

**Comment by neutrinoceros** [comment]
The previous comment https://github.com/astropy/astropy/issues/8651#issuecomment-2423925483 is still relevant to anyone who'd like to help here.

**Comment by tactipus** [comment]
lmao i had already commented on this. i was wondering why i got a notification for this issue

**Comment by tjb** [comment]
@neutrinoceros @astrofrog I made a quick PR with the added docstrings for version 7.0.0. happy to do more. i used tox to compile the the docs and make sure it was present. 

<img width="769" height="299" alt="Image" src="https://github.com/user-attachments/assets/912b3f15-5680-4ba9-9be4-a0dd21ee33ba" />

i read through the contrib documents and it mentions added to the `changes` directory...for this change, would directory should i add a change to? for each file should i add a changelog to that directory with something like "Added version docstring for X method for API documentation"? 

hope this is what everyone wanted! any feedback is very welcomed. happy to contribute to a v cool project :)

**Comment by Pittu-Sharma** [comment]
Hi, I would like to work on this issue.
Could you please assign it to me?

**Comment by LoveChauhan-18** [comment]
Hi, I would like to work on this issue. May I take it?

**Comment by neutrinoceros** [comment]
We do not assign issues. You are both free to contribute towards solving this one by opening pull requests, but I suggest you coordinate first to avoid repeated work. Thank you.

**Comment by Pittu-Sharma** [comment]
Hi @LoveChauhan-18,
Are you currently working on this issue?

If not, I would like to work on it.
Let’s coordinate to avoid duplicate work. Thanks!

**Comment by LoveChauhan-18** [comment]
@Pittu-Sharma , Sure you can work on it .

**Comment by shoaib-1122-tech** [comment]
Hi Everyone i am apply GSOC i have been exploring the astropy codebase and find issue is very interesting to start with..i would like to work on it could yo confirm it this is still open for contributtion if yes? i will start

**Comment by astrofrog** [comment]
I wonder whether it would be worth approaching this from the point of view of trying to create a tool to automate this somewhat, or at the very least to validate any versionadded tags - because checking these manually is going to be difficult for reviewers.

**Comment by pllim** [comment]
Maybe this is not really worth pursuing. Time is better spent on reviewing actual bug fixes. We will just have to remember to use this *going forward*.

**Comment by luvyarana** [comment]
hi i am a new contributor , shall i work on this issue ??

**Comment by Thenewbiecoder80221** [comment]
Hello! I’m a new contributor exploring Astropy through OpenAstronomy for GSoC.

I would like to work on this issue if it is still available. Please let me know if this would be a good starting point for a first contribution.

Thank you!
